app/controllers/RosmController.scala

```diff
@@ -35,15 +35,15 @@ trait RosmController extends LisaBaseController
   val get: Action[AnyContent] = Action.async { implicit request =>
     authorisedForLisa { (cacheId) =>
       hasAllSubmissionData(cacheId) { registrationDetails =>
-        rosmService.registerAndSubscribe(registrationDetails).flatMap {
+        rosmService.performSubscription(registrationDetails).flatMap {
           case Right(subscriptionId) => {
             Logger.info("Audit of Submission -> auditType = applicationReceived" + subscriptionId)
 
             auditService.audit(auditType = "applicationReceived",
               path = routes.RosmController.get().url,
               auditData = createAuditDetails(registrationDetails) ++ Map("subscriptionId" -> subscriptionId))
 
-            taxEnrolmentService.addSubscriber(subscriptionId, "SAFEID").map {
+            taxEnrolmentService.addSubscriber(subscriptionId, "subscriptionId").map {
```

Comment:
We need the actual safeId here for the second parameter. Unless that's covered by an upcoming task?